packages/eventindexer/repo/transaction_test.go

```diff
@@ -0,0 +1,103 @@
+package repo
+
+import (
+	"context"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/taikoxyz/taiko-mono/packages/eventindexer"
+	"github.com/taikoxyz/taiko-mono/packages/eventindexer/db"
+)
+
+func Test_NewTransactionRepo(t *testing.T) {
+	tests := []struct {
+		name    string
+		db      eventindexer.DB
+		wantErr error
+	}{
+		{
+			"success",
+			&db.DB{},
+			nil,
+		},
+		{
+			"noDb",
+			nil,
+			eventindexer.ErrNoDB,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			_, err := NewTransactionRepository(tt.db)
+			if err != tt.wantErr {
+				t.Errorf("NewTransactionRepository() error = %v, wantErr %v", err, tt.wantErr)
+				return
```

Comment:
tests do not need to return, and you dont need `t.Errorf`, you can use `assert.Equal(t, expectedError, err`